# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatform.cpp
# Commit: 0d746825f92a
# Full Hash: 0d746825f92a3039433aa275997c6e7f2ebfaf56
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Make InitFontList safe to call off-main-thread on macOS, and run font-list initialization on a separate thread during startup. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115295
# ==============================================================================

diff -r fbe7310ee5f3 -r 0d746825f92a gfx/thebes/gfxPlatform.cpp
--- a/gfx/thebes/gfxPlatform.cpp	Tue May 18 11:10:43 2021 +0000
+++ b/gfx/thebes/gfxPlatform.cpp	Tue May 18 11:10:43 2021 +0000
@@ -968,13 +968,11 @@
 
   gPlatform->mHasVariationFontSupport = gPlatform->CheckVariationFontSupport();
 
-  if (!StaticPrefs::gfx_font_list_lazy_init_enabled_AtStartup()) {
-    // On macOS we don't initialize the font list here because we want to allow
-    // more time for the RegisterFonts thread to complete. So it is initialized
-    // lazily on first use; see gfxPlatformFontList::PlatformFontList().
-    if (!gPlatform->CreatePlatformFontList()) {
-      MOZ_CRASH("Could not initialize gfxPlatformFontList");
-    }
+  // This *create* the platform font list instance, but may not *initialize* it
+  // yet if the gfx.font-list.lazy-init.enabled pref is set. The first *use*
+  // of the list will ensure it is initialized.
+  if (!gPlatform->CreatePlatformFontList()) {
+    MOZ_CRASH("Could not initialize gfxPlatformFontList");
   }
 
   gPlatform->mScreenReferenceSurface = gPlatform->CreateOffscreenSurface(